versioned/mongodb/src/main/java/com/dremio/nessie/versioned/impl/condition/BsonConditionExpression.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.versioned.impl.condition;
+
+import org.bson.conversions.Bson;
+
+/**
+ * This provides a separation of queries on @{ConditionExpression} from the object itself.
+ * This uses the Visitor design pattern to retrieve object attributes.
+ */
+public class BsonConditionExpression implements ConditionExpressionVisitor<Bson> {
+
+  /**
+   * Creates a BSON representation of the ConditionExpression object.
+   * @param conditionExpression the object to convert.
+   * @return BSON representation of conditionExpression
+   */
+  public Bson to(ConditionExpression conditionExpression) {
+    return conditionExpression.accept(this);
+  }
+
+   /**
+   * This is a callback method that ConditionExpression will call when this visitor is accepted.
+   * Its purpose is c reates a BSON representation of the ConditionExpression object.
```

Comment:
Nit: comment typo.

We also shouldn't expose internal implementation details in the comments, suggest you add a general purpose JavaDoc.

Reply:
Ok, reworded and moved this comment to a standard java comment. I think the design intent is still important.